(duplicate) partially (re-)implemented control messages

damanmkumar opened the pull request:
**NOTE:** duplicate of #46 since that is somewhat messed up, and shows unrelated changes
- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages IO channels around a socket
- (de)serializer classes manage (de)serialization based on recipient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- serialization and deserialization of `file_request`s and `chunk_request`s supported

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
Merge branch 'dev' into feat/ctl-msgs-alt

damanmkumar committed:
fixed wrong scoping of service + explicitly scoped
the type and value of {file|chunk}_request::msg_type

damanmkumar committed:
fixed wrong scoping of service + explicitly scoped
the type and value of {file|chunk}_request::msg_type

damanmkumar committed:
fixed wrong scoping of service + explicitly scoped
the type and value of {file|chunk}_request::msg_type

damanmkumar committed:
fixed wrong scoping of service + explicitly scoped
the type and value of {file|chunk}_request::msg_type

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused file

damanmkumar committed:
removed unused file

damanmkumar committed:
removed unused file

damanmkumar committed:
removed unused file

damanmkumar committed:
removed unused file

damanmkumar committed:
removed unused file

damanmkumar committed:
renamed {file|chunk}_request::msg_type to type

damanmkumar committed:
renamed {file|chunk}_request::msg_type to type

damanmkumar committed:
renamed {file|chunk}_request::msg_type to type

damanmkumar committed:
renamed {file|chunk}_request::msg_type to type

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
Merge branch 'feat/ctl-msgs-alt' of github.com:CS297P-SAD/sadfs into feat/ctl-msgs-alt

damanmkumar committed:
added deserialization support

damanmkumar committed:
added deserialization support

damanmkumar committed:
added deserialization support

damanmkumar committed:
Merge branch 'feat/ctl-msgs-alt' of github.com:CS297P-SAD/sadfs into feat/ctl-msgs-alt

damanmkumar commented:
https://github.com/CS297P-SAD/sadfs/pull/46#discussion_r532825344

abmono93 reviewed (changes requested):
Only comment is still the change from dealing with bytes to dealing with chunks.
Other than that.. I don't totally understand how everything in here works. But I verified that it does work on my machine

damanmkumar committed:
changed file_request to chunk_location_request

- the request now only contains a filename and chunk number

damanmkumar committed:
changed file_request to chunk_location_request

- the request now only contains a filename and chunk number

damanmkumar committed:
moved control messages into files based on intended recepient

damanmkumar committed:
moved control messages into files based on intended recepient

damanmkumar committed:
moved small function's definition to header file as an inline function definition

damanmkumar committed:
moved small function's definition to header file as an inline function definition

abmono93 committed:
Create chunkid and uuid (#55)

* Implemented add_chunk_to_file

* Changed internal protobuf

* changed file_chunks::at to file_chunks[]

* Formatting in internal.hpp

* implemented uuid class and fixed build error

* Removed test code

* Formatting sadmd.hpp

* Renamed functions to (de)serialize. Fixed range check for file_chunk[]. Optimizations.

* Fixed operator[] overload for file_chunks

* size_t -> std::size_t

* Changes in response to PR feedback

* Removed unused import stmt

* Fixed reference bug in load_file

* internal.hpp -> util.hpp

* Moved file_chunks into util namespace

* modified uuid IO to/from file_chunks (#57)

* modified uuid io to/from file_chunks

* included iterator header

* made uuid's size function constexpr static

* Revert "modified uuid IO to/from file_chunks (#57)" (#58)

This reverts commit 5b26365dc0ee4b5e20432c6a9c16e05a7caa2706.

Co-authored-by: Daman Mohan Kumar <kumardm@uci.edu>
Co-authored-by: damanmkumar <59807165+damanmkumar@users.noreply.github.com>

damanmkumar committed:
corrected old strings that referenced the file_request control message

damanmkumar committed:
corrected old strings that referenced the file_request control message

abmono93 committed:
Create chunkid alt - rest of code (#59)

* Implemented add_chunk_to_file

* Changed internal protobuf

* changed file_chunks::at to file_chunks[]

* Formatting in internal.hpp

* implemented uuid class and fixed build error

* Removed test code

* Formatting sadmd.hpp

* Renamed functions to (de)serialize. Fixed range check for file_chunk[]. Optimizations.

* Fixed operator[] overload for file_chunks

* size_t -> std::size_t

* Changes in response to PR feedback

* Removed unused import stmt

* Fixed reference bug in load_file

* internal.hpp -> util.hpp

* Moved file_chunks into util namespace

* saving test code

* modified uuid IO to/from file_chunks (#57)

* modified uuid io to/from file_chunks

* included iterator header

* made uuid's size function constexpr static

* changed from unsigned int to int

* Revert "modified uuid IO to/from file_chunks (#57)" (#58)

This reverts commit 5b26365dc0ee4b5e20432c6a9c16e05a7caa2706.

* switched create_file and load_file

* changed append_chunkid to take parameter. expiration_point->valid_until

* removed test code

* fixed build errors

Co-authored-by: Daman Mohan Kumar <kumardm@uci.edu>
Co-authored-by: damanmkumar <59807165+damanmkumar@users.noreply.github.com>

damanmkumar committed:
renamed control_message to message_container for better readability

damanmkumar committed:
renamed control_message to message_container for better readability

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
added deserialization support

damanmkumar committed:
Merge branch 'feat/ctl-msgs-alt' of github.com:CS297P-SAD/sadfs into feat/ctl-msgs-alt

damanmkumar committed:
fixed git's rebase mess up

damanmkumar committed:
more cleaning up after git

damanmkumar commented:
`git rebase` gone wrong. This PR is beyond help.
